packages/@ember/-internals/views/lib/system/event_dispatcher.js

```diff
@@ -360,6 +360,8 @@ export default EmberObject.extend({
                 event.preventDefault();
                 event.stopPropagation();
                 break;
+              } else if (event.cancelBubble === true) {
+                break;
```

Comment:
This is probably not the correct nesting for this. I believe we want to check the flag after the event handler has run (currently the check runs when if there is no event handler). Further, we probably need the same treatment for the action branch too. So I think you probably want to do it before setting the target to the parent.

I feel like this should have failed the test tho?

Reply:
Also, we should make sure to mirror the jquery/dom behavior that stop propagation doesn’t prevent handlers on the same element from firing

Reply:
On that note, we probably have another bug here. It seems like the code assumes you can either have a component element handler, or action handlers, but not both. I believe with the possibility of forwarding action modifiers through splattributes, this is no longer a correct assumption.

Reply:
> I believe we want to check the flag after the event handler has run

In what scenario would that not be the case? `viewHandler(target, event)` will always be called before our `cancelBubble` check, right? 

> Further, we probably need the same treatment for the action branch too.

I tried to add a (failing) test for that, until I realized that the action handler will not receive a reference to the event, so cannot call `stopPropagation()`, or do I miss something?

> stop propagation doesn’t prevent handlers on the same element from firing

How would there be two different handlers (component event handlers) on the same element? You mean the case you mentioned in your following comment? 👇

> It seems like the code assumes you can either have a component element handler, or action handlers, but not both. I believe with the possibility of forwarding action modifiers through splattributes, this is no longer a correct assumption.

Hm, I see... 🤔Good catch! But we should tackle this in a separate PR I think?

Reply:
> `viewHandler(target, event)` will always be called before our `cancelBubble` check, right?

You're right!


> action handler will not receive a reference to the event

Don't we pass it [here](https://github.com/emberjs/ember.js/blob/561759e4d3cd2f978deb057e4e1f52d03098f40d/packages/%40ember/-internals/views/lib/system/event_dispatcher.js#L292)?

> How would there be two different handlers (component event handlers) on the same element? You mean the case you mentioned in your following comment? 👇

Yep that's one case I could think of. I think you could also do `<div {{action "foo"}} {{action "bar"}} />`? Does that work? (Looking at the code, I'm not sure where this is handled.)

> But we should tackle this in a separate PR I think?

Seems fine to me!

Reply:
> Don't we pass it here?

Yes, but that goes to https://github.com/emberjs/ember.js/blob/83263728b116fac6d4e8fcc798babf555a62ac2b/packages/%40ember/-internals/glimmer/lib/modifiers/action.ts#L125-L181, and there the event is not passed to the action handler. I believe this is how the action helper API works: the (element space(!), not closure) action helper passes only the custom args, not the event itself. That's why we have things like `bubbles=false`.

>  I think you could also do <div {{action "foo"}} {{action "bar"}} />? Does that work? (Looking at the code, I'm not sure where this is handled.)

We have that loop here: https://github.com/emberjs/ember.js/blob/561759e4d3cd2f978deb057e4e1f52d03098f40d/packages/%40ember/-internals/views/lib/system/event_dispatcher.js#L288

And a related test here, though for different events: https://github.com/emberjs/ember.js/blob/561759e4d3cd2f978deb057e4e1f52d03098f40d/packages/%40ember/-internals/glimmer/tests/integration/helpers/element-action-test.js#L1323-L1360

I tried to add a similar test with the same event type, but that fails, as the first handled action handler returns early here: https://github.com/emberjs/ember.js/blob/561759e4d3cd2f978deb057e4e1f52d03098f40d/packages/%40ember/-internals/views/lib/system/event_dispatcher.js#L292

So this seems to be not supported at this point. I guess it should be possible, by always looping over all action handlers, and return `false` if at least one returned `false`!?

Reply:
Oh, wait a second. Actually that test with multiple `{{action}}` *does*  pass w/ jQuery, so it seems it should actually work, but doesn't in no-jQuery mode! 

I'll bring up another PR to fix this specific issue. As for the issue in this PR, I *think* this PR should be ok as it is?

Reply:
Here we go: #17876